app/services/ProbationOffenderSearch.java

```diff
@@ -0,0 +1,185 @@
+                                        .probationAreaCodes(bearerToken)),
+                                "total", body.get("totalElements"),
+                                "suggestions", body.get("suggestions")
+                        ))
+                        .thenCompose(body -> withProbationAreaDescriptions(bearerToken, body))
+                        .thenApply(body -> withPrisonImageLinkCodes(bearerToken, body))
+                );
+
+    }
+
+    private CompletionStage<Map<String, Object>> withProbationAreaDescriptions(String bearerToken, ImmutableMap<String, Object> body) {
+        val aggregationsWrapper = aggregationsWrapper(body);
+        return offenderApi.probationAreaDescriptions(bearerToken,
+                extractProbationAreaCodes(aggregationsWrapper))
+                .thenApply(areas -> ImmutableMap.of(
+                        "offenders", body.get("offenders"),
+                        "aggregations", addProbationAreaDescription(aggregationsWrapper, areas),
+                        "total", body.get("total"),
+                        "suggestions", body.get("suggestions")
+                ));
+    }
+
+    @SuppressWarnings("unchecked")
+    private Map<String, Object> withPrisonImageLinkCodes(String bearerToken, Map<String, Object> body) {
+        final Function<String, String> oneTimeNomisRef = nomsNumber -> OffenderController
+                .generateOneTimeImageReference(encrypter, nomsNumber, bearerToken);
+
+        val offenders = (List<Map<String, Object>>) body.get("offenders");
+        offenders.forEach(offender -> Optional.ofNullable(offender.get("otherIds"))
+                .flatMap(otherIds -> Optional.ofNullable(((Map<String, Object>) otherIds).get("nomsNumber")))
+                .ifPresent(nomsNumber -> offender
+                        .put("oneTimeNomisRef", oneTimeNomisRef.apply(nomsNumber.toString()))));
+        return body;
+    }
+
+    @Override
+    public CompletionStage<HealthCheckResult> isHealthy() {
+        return wsClient.url(apiBaseUrl + "health/ping").
+                get().
+                thenApply(wsResponse -> {
+                    val healthy = wsResponse.getStatus() == OK;
+                    if (!healthy) {
+                        Logger.warn("probation offender search response status: " + wsResponse.getStatus());
+                        return unhealthy(String.format("Status %d", wsResponse.getStatus()));
+                    }
+                    return healthy(wsResponse.asJson());
+                }).
+                exceptionally(throwable -> {
+                    Logger.error("Error while checking probation offender search connectivity", throwable);
+                    return unhealthy(throwable.getLocalizedMessage());
+                });
+    }
+
+    @Override
+    public CompletionStage<MatchedOffenders> findMatch(String bearerToken, CourtDefendant offender) {
```

Comment:
As comment, these methods are not used "in anger" - they were part of a spike for Prepare a Case that was never used. The code should really be removed